controller/destination/listener.go

```diff
@@ -85,9 +86,13 @@ func (l *endpointListener) toWeightedAddrSet(endpoints []common.TcpAddress) *pb.
 			if len(resultingPods) == 0 || resultingPods[0] == nil {
```

Comment:
TIOLI: since we're no longer referencing `resultingPods[0]`, I think it makes sense to remove this check as well. I'm also not sure why we had a nil check in the first place, since I wouldn't expect `GetPodsByIndex` to return a slice containing `nil` elements. Anyway, I'd probably just move this log line to after the `for` loop, in the event that we didn't find a running pod. Something like:

```
		} else {
			podFound := false
			for _, pod := range resultingPods {
				if pod.Status.Phase == coreV1.PodRunning {
					podFound = true
					metricLabelsForPod = pkgK8s.GetOwnerLabels(pod.ObjectMeta)
					metricLabelsForPod["pod"] = pod.Name
					break
				}
			}
			if !podFound {
				log.Errorf("Could not find running pod for IP [%s], this IP will be sent with no metric labels.", ipAsString)
			}
		}
```